openfasoc/generators/glayout/glayout/flow/blocks/composite/regulated_cascoded_current_mirror/regulated_cascoded_current_mirror.py

```diff
@@ -0,0 +1,374 @@
+from glayout.flow.routing.c_route import c_route
+from glayout.flow.routing.L_route import L_route
+
+from glayout.flow.placement.two_transistor_interdigitized import two_nfet_interdigitized, two_pfet_interdigitized
+
+from glayout.flow.pdk.util.snap_to_grid import component_snap_to_grid
+from glayout.flow.pdk.util.comp_utils import evaluate_bbox, prec_center, prec_array, movey, align_comp_to_port, prec_ref_center
+from glayout.flow.pdk.util.port_utils import rename_ports_by_orientation, rename_ports_by_list, add_ports_perimeter, print_ports, set_port_orientation, rename_component_ports
+
+from gdsfactory.components import text_freetype, rectangle
+from gdsfactory import Component
+from gdsfactory.routing.route_quad import route_quad
+from glayout.flow.spice.netlist import Netlist
+from typing import Optional, Union 
+
+def delete_files_in_directory(directory_path):
+   try:
+     files = listdir(directory_path)
+     for file in files:
+       file_path = path.join(directory_path, file)
+       if path.isfile(file_path):
+         try:
+            chmod(file_path, 0o777)
+            remove(file_path)
+         except PermissionError:
+               print(f"PermissionError: Operation not permitted for {file_path}")
+     print("All files deleted successfully.")
+   except OSError:
+     print("Error occurred while deleting files.")
+
+global PDK_ROOT
+if 'PDK_ROOT' in environ:
+	PDK_ROOT = str(Path(environ['PDK_ROOT']).resolve())
+else:
+	PDK_ROOT = "/usr/bin/miniconda3/share/pdk/"
+ 
+def generate_current_mirror_netlist(
+    pdk: MappedPDK,
+    instance_name: str,
+    CM_size: tuple[float, float, int, int],  # (width, length, multipliers, fingers)
+    drain_net_A: str,
+    gate_net: str,
+    source_net: str,
+    drain_net_B: str,
+    transistor_type: str = "nfet",
+    bulk_net: str = None,
+    proposed_ground: str = None,  # Proposed ground net
+    dummy: bool = True,
+    subckt_only: bool = False,
+) -> Netlist:
+    """Generate a netlist for a current mirror."""
+
+    if bulk_net is None:
+        bulk_net = "VDD" if transistor_type.lower() == "pfet" else "VSS"
+
+    width = CM_size[0]
+    length = CM_size[1]
+    multipliers = CM_size[2]  
+    mtop = multipliers if subckt_only else 1
+    #mtop = multipliers * 2 if dummy else multipliers # Double the multiplier to account for the dummies
+    fingers =  CM_size[3] # Number of fingers of the interdigitized fets
+
+
+    model_name = pdk.models[transistor_type.lower()]
+
+    circuit_name = instance_name
+    nodes = [drain_net_A, drain_net_B, proposed_ground]
+
+    source_netlist = f".subckt {circuit_name} {' '.join(nodes)}\n"
+
+    #source_netlist += f"V{proposed_ground}1 ({proposed_ground} {bulk_net}) 0\n" #Proposed ground connection
+
+
+    # Generating only two transistors (one on each side):
+    source_netlist += f"XA {drain_net_A} {gate_net} {source_net} {bulk_net} {model_name} l={length} w={width} m={mtop} nf={fingers}\n"
```

Comment:
There is no need to write spice to generate the netlist, it can be done programmatically. Refer to the opamp netlist generation code.